doc strings are missing in lot of modules.

While checking with pylint, it gives a lot of suggestions to add doc strings in the functions.  
```************* Module tests.test_common
tests/test_common.py:1:0: C0111: Missing module docstring (missing-docstring)
tests/test_common.py:6:0: E0401: Unable to import 'PIL' (import-error)
tests/test_common.py:19:0: C0111: Missing function docstring (missing-docstring)
tests/test_common.py:24:0: C0111: Missing function docstring (missing-docstring)
tests/test_common.py:29:0: C0111: Missing function docstring (missing-docstring)
tests/test_common.py:34:0: C0111: Missing function docstring (missing-docstring)
tests/test_common.py:39:0: C0111: Missing function docstring (missing-docstring)
tests/test_common.py:44:0: C0111: Missing function docstring (missing-docstring)
tests/test_common.py:49:0: C0111: Missing function docstring (missing-docstring)```

docString are not that important for test functions. Its not just tests that is missing function docString.

```************* Module kodi_addon_checker.reporter
kodi_addon_checker/reporter.py:44:0: C0301: Line too long (120/100) (line-too-long)
kodi_addon_checker/reporter.py:13:0: C0111: Missing class docstring (missing-docstring)
kodi_addon_checker/reporter.py:15:4: C0111: Missing method docstring (missing-docstring)
kodi_addon_checker/reporter.py:13:0: R0903: Too few public methods (1/2) (too-few-public-methods)
kodi_addon_checker/reporter.py:19:0: C0111: Missing class docstring (missing-docstring)
kodi_addon_checker/reporter.py:19:0: R0205: Class 'ReportManager' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
kodi_addon_checker/reporter.py:23:4: C0111: Missing method docstring (missing-docstring)
kodi_addon_checker/reporter.py:37:4: C0103: Method name "getEnabledReporters" doesn't conform to snake_case naming style (invalid-name)
kodi_addon_checker/reporter.py:37:4: C0111: Missing method docstring (missing-docstring)
kodi_addon_checker/reporter.py:41:0: C0111: Missing function docstring (missing-docstring)```

--- a/kodi_addon_checker/__main__.py
+++ b/kodi_addon_checker/__main__.py
@@ -7,15 +7,15 @@
 """
 
 import argparse
+import logging
 import os
 import sys
-import logging
-from kodi_addon_checker import logger
-from kodi_addon_checker import check_addon
+
+from kodi_addon_checker import check_addon, logger
 from kodi_addon_checker.check_repo import check_repo
 from kodi_addon_checker.common import load_plugins
-from kodi_addon_checker.config import ConfigManager, Config
-from kodi_addon_checker.record import Record, PROBLEM, WARNING, INFORMATION
+from kodi_addon_checker.config import Config, ConfigManager
+from kodi_addon_checker.record import INFORMATION, PROBLEM, WARNING, Record
 from kodi_addon_checker.report import Report
 from kodi_addon_checker.reporter import ReportManager
 

--- a/kodi_addon_checker/addons/Addon.py
+++ b/kodi_addon_checker/addons/Addon.py
@@ -6,9 +6,9 @@
     See LICENSES/README.md for more information.
 """
 
-from .AddonDependency import AddonDependency
-
 import xml.etree.ElementTree as ET
+
+from .AddonDependency import AddonDependency
 
 
 class Addon(object):

--- a/kodi_addon_checker/addons/AddonDependency.py
+++ b/kodi_addon_checker/addons/AddonDependency.py
@@ -6,8 +6,8 @@
     See LICENSES/README.md for more information.
 """
 
-from distutils.version import LooseVersion
 import xml.etree.ElementTree as ET
+from distutils.version import LooseVersion
 
 
 class AddonDependency(object):

--- a/kodi_addon_checker/addons/Repository.py
+++ b/kodi_addon_checker/addons/Repository.py
@@ -6,12 +6,13 @@
     See LICENSES/README.md for more information.
 """
 
-from .Addon import Addon
-
 import gzip
-import requests
 import xml.etree.ElementTree as ET
 from io import BytesIO
+
+import requests
+
+from .Addon import Addon
 
 
 class Repository(object):

--- a/kodi_addon_checker/check_addon.py
+++ b/kodi_addon_checker/check_addon.py
@@ -30,6 +30,14 @@ LOGGER = logging.getLogger(__name__)
 
 
 def start(addon_path, args, all_repo_addons, config=None):
+    """Starting point of all the checks that
+       are to be performed on an addon
+
+       :addon_path: Path to the addon that is to be checked
+       :args: argparse object
+       :all_repo_addons: a nested list having information
+                         about all the repo addons
+    """
     addon_id = os.path.basename(os.path.normpath(addon_path))
     addon_report = Report(addon_id)
     LOGGER.info("Checking add-on %s" % addon_id)

--- a/kodi_addon_checker/check_addon.py
+++ b/kodi_addon_checker/check_addon.py
@@ -6,24 +6,17 @@
     See LICENSES/README.md for more information.
 """
 
+import logging
 import os
 import xml.etree.ElementTree as ET
-import logging
 
+from . import (check_artwork, check_dependencies, check_entrypoint,
+               check_files, check_old_addon, check_py3_compatibility,
+               check_string, check_url, common, handle_files,
+               schema_validation)
 from .addons.Repository import Repository
-from .record import Record, INFORMATION
+from .record import INFORMATION, Record
 from .report import Report
-from . import check_artwork
-from . import check_old_addon
-from . import check_dependencies
-from . import check_entrypoint
-from . import handle_files
-from . import check_files
-from . import check_string
-from . import check_py3_compatibility
-from . import check_url
-from . import common
-from . import schema_validation
 
 ROOT_URL = "http://mirrors.kodi.tv/addons/{branch}/addons.xml.gz"
 LOGGER = logging.getLogger(__name__)

--- a/kodi_addon_checker/check_artwork.py
+++ b/kodi_addon_checker/check_artwork.py
@@ -6,13 +6,15 @@
     See LICENSES/README.md for more information.
 """
 
+import logging
 import os
 import re
-import logging
+
 from PIL import Image
-from .report import Report
+
 from .common import has_transparency, relative_path
-from .record import PROBLEM, Record, WARNING, INFORMATION
+from .record import INFORMATION, PROBLEM, WARNING, Record
+from .report import Report
 
 LOGGER = logging.getLogger(__name__)
 

--- a/kodi_addon_checker/check_artwork.py
+++ b/kodi_addon_checker/check_artwork.py
@@ -106,6 +106,12 @@ def _assests(image_type: str, parsed_xml, addon_path: str):
 
 
 def _check_icon(report: Report, im, width, height):
+    """Check the icon of the addon for transparency and dimensions
+
+        :im: PIL.Image object
+        :width: width of the icon
+        :height: height of the icon
+    """
     if has_transparency(im):
         report.add(Record(PROBLEM, "Icon.png should be solid. It has transparency."))
 
@@ -120,6 +126,11 @@ def _check_icon(report: Report, im, width, height):
 
 
 def _check_fanart(report: Report, width, height):
+    """Check the dimensions of the fanart
+
+        :width: width of the icon
+        :height: height of the icon
+    """
     fanart_sizes = [(1280, 720), (1920, 1080), (3840, 2160)]
     fanart_sizes_str = " or ".join(["%dx%d" % (w, h) for w, h in fanart_sizes])
 

--- a/kodi_addon_checker/check_dependencies.py
+++ b/kodi_addon_checker/check_dependencies.py
@@ -6,13 +6,12 @@
     See LICENSES/README.md for more information.
 """
 
-from .addons.Addon import Addon
-
 import logging
 from distutils.version import LooseVersion
 
+from .addons.Addon import Addon
+from .record import INFORMATION, PROBLEM, WARNING, Record
 from .report import Report
-from .record import PROBLEM, Record, WARNING, INFORMATION
 
 common_ignore_deps = ['xbmc.metadata.scraper.albums', 'xbmc.metadata.scraper.movies',
                       'xbmc.metadata.scraper.musicvideos', 'xbmc.metadata.scraper.tvshows',

--- a/kodi_addon_checker/check_dependencies.py
+++ b/kodi_addon_checker/check_dependencies.py
@@ -98,6 +98,14 @@ def check_addon_dependencies(report: Report, repo_addons: dict, parsed_xml, bran
 
 
 def check_reverse_dependencies(report: Report, addon: str, branch_name: str, all_repo_addons: dict):
+    """Check for orphaned addon i.e the addons that are not requirement of any other addon.
+
+      :addon: addon that is to be checked
+      :branch_name: name of the branch on which the
+                    checks are being performed
+      :all_repo_addons: a nested list having information
+                        about all the repo addonst
+    """
     addonInRepo = None
     rdepends = []
     rdependsLowerBranch = []
@@ -133,6 +141,9 @@ def check_reverse_dependencies(report: Report, addon: str, branch_name: str, all
 
 
 def _get_ignore_list(branch_name):
+    """Generate an dependency ignore list based
+       on the branch name
+    """
 
     if branch_name == "leia":
         common_ignore_deps.extend(["script.module.pycryptodome"])
@@ -151,7 +162,6 @@ def _check_extensions(report: Report, parsed_xml, addon):
        the existing extension points
 
       :addon: class kodi_addon_checker.addons.Addon.Addon
-
     """
     deps = [dependency.id for dependency in addon.dependencies]
 

--- a/kodi_addon_checker/check_entrypoint.py
+++ b/kodi_addon_checker/check_entrypoint.py
@@ -7,9 +7,11 @@
 """
 
 import os
+
 from radon.raw import analyze
+
+from .record import PROBLEM, WARNING, Record
 from .report import Report
-from .record import PROBLEM, Record, WARNING
 
 
 def check_complex_addon_entrypoint(report: Report, addon_path: str, parsed_xml, max_entrypoint_count: int):

--- a/kodi_addon_checker/check_files.py
+++ b/kodi_addon_checker/check_files.py
@@ -6,14 +6,15 @@
     See LICENSES/README.md for more information.
 """
 
+import json
 import os
 import re
-import json
 import xml.etree.ElementTree as ET
 from pathlib import Path
+
 from . import handle_files
 from .common import relative_path
-from .record import PROBLEM, Record, WARNING, INFORMATION
+from .record import INFORMATION, PROBLEM, WARNING, Record
 from .report import Report
 
 

--- a/kodi_addon_checker/check_old_addon.py
+++ b/kodi_addon_checker/check_old_addon.py
@@ -43,6 +43,14 @@ def _get_addon_name(xml_path: str):
 
 
 def _check_versions(report: Report, addon_name, branch, addon_version, repo_addons_version, pr):
+    """Check for version bump in the existing addon
+
+        :addon_name:         addon that is to be checked
+        :addon_version:      the version of addon that is submitted
+        :repo_addon_version: version of addon present in Kodi repository
+        :pr:                 boolean value indicating whether the check is
+                             running on pull request or not
+    """
     if pr:
         if LooseVersion(addon_version) > LooseVersion(repo_addons_version):
             LOGGER.info("%s addon have greater version: %s than repo_version: %s in branch %s"

--- a/kodi_addon_checker/check_old_addon.py
+++ b/kodi_addon_checker/check_old_addon.py
@@ -6,12 +6,13 @@
     See LICENSES/README.md for more information.
 """
 
-import os
 import logging
+import os
 import xml.etree.ElementTree as ET
 from distutils.version import LooseVersion
+
+from .record import INFORMATION, PROBLEM, Record
 from .report import Report
-from .record import PROBLEM, Record, INFORMATION
 
 LOGGER = logging.getLogger(__name__)
 

--- a/kodi_addon_checker/check_py3_compatibility.py
+++ b/kodi_addon_checker/check_py3_compatibility.py
@@ -6,15 +6,13 @@
     See LICENSES/README.md for more information.
 """
 
-import os
-
-from lib2to3 import pgen2
-from lib2to3 import refactor
 import difflib
+import os
+from lib2to3 import pgen2, refactor
 
 from .common import relative_path
+from .record import INFORMATION, PROBLEM, Record
 from .report import Report
-from .record import Record, INFORMATION, PROBLEM
 
 
 class KodiRefactoringTool(refactor.RefactoringTool):

--- a/kodi_addon_checker/check_repo.py
+++ b/kodi_addon_checker/check_repo.py
@@ -13,6 +13,14 @@ from .report import Report
 
 
 def check_repo(repo_path, args, all_repo_addons, config):
+    """Perform all the check on a complete repository
+
+        :repo_path: Path of the repo that is to be tested
+        :args: argparse object
+        :all_repo_addons: a nested list having information
+                          about all the repo addons
+        :config: Config object
+    """
     repo_report = Report(repo_path)
     repo_report.add(Record(INFORMATION, "Checking repository %s" % repo_path))
     toplevel_folders = sorted(next(os.walk(repo_path))[1])

--- a/kodi_addon_checker/check_repo.py
+++ b/kodi_addon_checker/check_repo.py
@@ -7,7 +7,9 @@
 """
 
 import os
+
 import kodi_addon_checker.check_addon as check_addon
+
 from .record import INFORMATION, PROBLEM, Record
 from .report import Report
 

--- a/kodi_addon_checker/check_string.py
+++ b/kodi_addon_checker/check_string.py
@@ -7,12 +7,14 @@
 """
 
 import os
-import polib
 import re
-from .report import Report
-from .common import relative_path
+
+import polib
+
 from . import handle_files
-from .record import INFORMATION, PROBLEM, Record, WARNING
+from .common import relative_path
+from .record import INFORMATION, PROBLEM, WARNING, Record
+from .report import Report
 
 
 def check_for_legacy_strings_xml(report: Report, addon_path: str):

--- a/kodi_addon_checker/check_url.py
+++ b/kodi_addon_checker/check_url.py
@@ -6,11 +6,11 @@
     See LICENSES/README.md for more information.
 """
 
-from .record import Record, WARNING
-from .report import Report
-
-import urllib3
 import requests
+import urllib3
+
+from .record import WARNING, Record
+from .report import Report
 
 
 def check_url(report: Report, parsed_xml):

--- a/kodi_addon_checker/common.py
+++ b/kodi_addon_checker/common.py
@@ -15,6 +15,10 @@ REL_PATH = ""
 
 
 def has_transparency(im):
+    """Check the transparency(aplha layer) in the given image
+
+        :im: PIL.Image object
+    """
     try:
         if im.mode == "RGBA":
             alpha = im.split()[-1]
@@ -30,6 +34,8 @@ def has_transparency(im):
 
 
 def load_plugins():
+    """Load the reporter plugins
+    """
     plugins_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "plugins")
     sys.path.append(plugins_dir)
     for _, package_name, _ in pkgutil.iter_modules([plugins_dir]):
@@ -38,5 +44,7 @@ def load_plugins():
 
 
 def relative_path(file_path):
+    """Return the relative_path of the file
+    """
     path_to_print = file_path[len(REL_PATH):]
     return ".{}".format(path_to_print)

--- a/kodi_addon_checker/handle_files.py
+++ b/kodi_addon_checker/handle_files.py
@@ -6,12 +6,13 @@
     See LICENSES/README.md for more information.
 """
 
+import os
 import pathlib
 import re
-import os
-from .report import Report
+
 from .common import relative_path
 from .record import PROBLEM, Record
+from .report import Report
 
 
 def find_file(name: str, path: str):

--- a/kodi_addon_checker/logger.py
+++ b/kodi_addon_checker/logger.py
@@ -14,7 +14,8 @@ class Logger:
 
     @staticmethod
     def create_logger(debug_filename, logger_name):
-
+        """Creates a logger format for error logging
+        """
         logger = logging.getLogger(logger_name)
         logger.setLevel(logging.DEBUG)
         logger.propagate = False

--- a/kodi_addon_checker/plugins/console_reporter.py
+++ b/kodi_addon_checker/plugins/console_reporter.py
@@ -6,7 +6,7 @@
     See LICENSES/README.md for more information.
 """
 
-from kodi_addon_checker.record import INFORMATION, WARNING, PROBLEM
+from kodi_addon_checker.record import INFORMATION, PROBLEM, WARNING
 from kodi_addon_checker.report import Record
 from kodi_addon_checker.reporter import Reporter, reporter
 
@@ -23,7 +23,8 @@ def colorPrint(string, color):
 
 @reporter(name="console", enabled=True)
 class ConsoleReporter(Reporter):
-
+    """Present Report on the console
+    """
     def report(self, report):
         if type(report) is Record:
             if report.log_level == INFORMATION:

--- a/kodi_addon_checker/report.py
+++ b/kodi_addon_checker/report.py
@@ -6,7 +6,7 @@
     See LICENSES/README.md for more information.
 """
 
-from .record import Record, PROBLEM, WARNING
+from .record import PROBLEM, WARNING, Record
 from .reporter import ReportManager
 
 

--- a/kodi_addon_checker/schema_validation.py
+++ b/kodi_addon_checker/schema_validation.py
@@ -6,13 +6,14 @@
     See LICENSES/README.md for more information.
 """
 
-from os.path import abspath, dirname, join, exists, split
 import copy
 import logging
+from os.path import abspath, dirname, exists, join, split
+
 import xmlschema
 
+from .record import INFORMATION, PROBLEM, Record
 from .report import Report
-from .record import PROBLEM, Record, INFORMATION
 
 XML_SCHEMA = join(dirname(__file__), 'xml_schema')
 LOGGER = logging.getLogger(__name__)

